ui/seek_bar.js

```diff
@@ -120,6 +163,97 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
           this.onAdCuePointsChanged_();
         });
 
+    this.eventManager.listen(this.bar, 'mousemove', (event) => {
+      if (!this.player.getImageTracks().length) {
+        this.hideThumbnail_();
+        return;
+      }
+      const rect = this.bar.getBoundingClientRect();
+      const min = parseFloat(this.bar.min);
+      const max = parseFloat(this.bar.max);
+      // Pixels from the left of the range element
+      const mousePosition = event.clientX - rect.left;
+      // Pixels per unit value of the range element.
+      const scale = (max - min) / rect.width;
+      // Mouse position in units, which may be outside the allowed range.
+      const value = Math.round(min + scale * mousePosition);
+      // Show Thumbnail
+      this.showThumbnail_(mousePosition, value);
+    });
+
+    this.eventManager.listen(this.bar, 'keydown', (event) => {
+      this.hideThumbnailTimer_.stop();
```

Comment:
It's not necessary to put in new event listeners for keyboard events, because `controls.js` already tracks those.

You can put the code that sets `this.isMoving_ = true` into the `onChangeStart` handler in this class, and the code that shows the thumbnail into `onChange`. The existing code will handle changing the value of the bar.

This has the advantage that it also handles bar movements that aren't based on keyboard left and right, such as pressing "start" or "end".

Reply:
Done!